lib/phoenix/socket/transport.ex

```diff
@@ -365,33 +365,39 @@ defmodule Phoenix.Socket.Transport do
     end)
   end
 
-  defp parse_origin(origin) do
+  defp parse_origin("http://*." <> orig),  do: parse_origin("http://" <> orig, :wildcard)
+  defp parse_origin("https://*." <> orig), do: parse_origin("https://" <> orig, :wildcard)
+  defp parse_origin("//*." <> orig),       do: parse_origin("//" <> orig, :wildcard)
+  defp parse_origin(origin, type \\ :static) do
```

Comment:
Good call. It was mainly due to laziness, but I'll add :static directly in the map